navigation/navigation-compose-material/src/androidTest/java/androidx/navigation/compose/material/NavGraphBuilderTest.kt

```diff
@@ -0,0 +1,171 @@
+                bottomSheet("$secondRoute/{$key}") { }
+            }
+        }
+
+        composeTestRule.runOnUiThread {
+            navController.navigate("$secondRoute/$arg")
+            assertThat(navController.currentBackStackEntry!!.arguments!!.getString(key))
+                .isEqualTo(arg)
+        }
+    }
+
+    @Test
+    fun testDefaultArguments() {
+        lateinit var navController: TestNavHostController
+        val key = "key"
+        val defaultArg = "default"
+        composeTestRule.setContent {
+            navController = TestNavHostController(LocalContext.current)
+            navController.navigatorProvider.addNavigator(createBottomSheetNavigator())
+
+            NavHost(navController, startDestination = firstRoute) {
+                bottomSheet(firstRoute) { }
+                bottomSheet(
+                    secondRoute,
+                    arguments = listOf(navArgument(key) { defaultValue = defaultArg })
+                ) { }
+            }
+        }
+
+        composeTestRule.runOnUiThread {
+            navController.navigate(secondRoute)
+            assertThat(navController.currentBackStackEntry!!.arguments!!.getString(key))
+                .isEqualTo(defaultArg)
+        }
+    }
+
+    @Test
+    fun testDeepLink() {
+        lateinit var navController: TestNavHostController
+        val uriString = "https://www.example.com"
+        val deeplink = NavDeepLinkRequest.Builder.fromUri(Uri.parse(uriString)).build()
+        composeTestRule.setContent {
+            navController = TestNavHostController(LocalContext.current)
+            navController.navigatorProvider.addNavigator(createBottomSheetNavigator())
+
+            NavHost(navController, startDestination = firstRoute) {
+                bottomSheet(firstRoute) { }
+                bottomSheet(
+                    secondRoute,
+                    deepLinks = listOf(navDeepLink { uriPattern = uriString })
+                ) { }
+            }
+        }
+
+        composeTestRule.runOnUiThread {
+            navController.navigate(uriString.toUri())
+            assertThat(navController.currentBackStackEntry!!.destination.hasDeepLink(deeplink))
+                .isTrue()
+        }
+    }
+
+    @Test
+    fun testNavigationNestedStart() {
+        lateinit var navController: TestNavHostController
+        composeTestRule.setContent {
+            navController = TestNavHostController(LocalContext.current)
+            navController.navigatorProvider.addNavigator(createBottomSheetNavigator())
+
+            NavHost(navController, startDestination = firstRoute) {
+                navigation(startDestination = secondRoute, route = firstRoute) {
+                    bottomSheet(secondRoute) { }
+                }
+            }
+        }
+
+        composeTestRule.runOnUiThread {
+            assertWithMessage("Destination should be added to the graph")
+                .that(firstRoute in navController.graph)
```

Comment:
`firstRoute` is the route of the navigation graph and not the `bottomSheet` destination itself - don't we want to check on the existence of `secondRoute`? That seems to imply that this check should be something like `secondRoute in (navController.graph.findNode(firstRoute) as NavGraph)`?

Reply:
Yep good catch!